src/Shared/UnitTests/EngineTestEnvironment.cs

```diff
@@ -4,12 +4,36 @@
 using System.Collections.Generic;
 using System.IO;
 using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
 using Shouldly;
 
 namespace Microsoft.Build.UnitTests
 {
     public partial class TestEnvironment
     {
+        // reset the default build manager and the state it might have accumulated from other tests
+        private object _resetBuildManager = new ResetDefaultBuildManager();
+
+        private class ResetDefaultBuildManager
+        {
+            public ResetDefaultBuildManager()
```

Comment:
@rainersigwald Are you OK with this preventive cleanup? It does not seem to have a noticeable impact on test execution time.